logging/net_appender.go

```diff
@@ -210,7 +210,7 @@ func (nl *NetAppender) backgroundWorker() {
 		if !utils.SelectContextOrWait(nl.cancelCtx, interval) {
 			cancelled = true
 		}
-		err := nl.Sync()
```

Comment:
Keeping sync internal works.

If we do need to re-expose sync (I kind of hope we don't), it's customary to just have sync take a mutex. One can't really do better than that.